Analysis/Tasks/PWGHF/HFTreeCreatorD0ToKPi.cxx

```diff
@@ -0,0 +1,263 @@
+  if (candidate.FunctionSelection() >= 1) {                                                                             \
+    rowCandidateFull(                                                                                                   \
+      candidate.index0_as<aod::BigTracksPID>().collision().bcId(),                                                      \
+      candidate.index0_as<aod::BigTracksPID>().collision().numContrib(),                                                \
+      candidate.posX(),                                                                                                 \
+      candidate.posY(),                                                                                                 \
+      candidate.posZ(),                                                                                                 \
+      candidate.xSecondaryVertex(),                                                                                     \
+      candidate.ySecondaryVertex(),                                                                                     \
+      candidate.zSecondaryVertex(),                                                                                     \
+      candidate.errorDecayLength(),                                                                                     \
+      candidate.errorDecayLengthXY(),                                                                                   \
+      candidate.chi2PCA(),                                                                                              \
+      candidate.rSecondaryVertex(),                                                                                     \
+      candidate.decayLength(),                                                                                          \
+      candidate.decayLengthXY(),                                                                                        \
+      candidate.decayLengthNormalised(),                                                                                \
+      candidate.decayLengthXYNormalised(),                                                                              \
+      candidate.impactParameterNormalised0(),                                                                           \
+      TMath::Abs(candidate.ptProng0()),                                                                                 \
+      TMath::Sqrt(RecoDecay::P(candidate.pxProng0(), candidate.pyProng0(), candidate.pzProng0())),                      \
+      candidate.impactParameterNormalised1(),                                                                           \
+      TMath::Abs(candidate.ptProng1()),                                                                                 \
+      TMath::Sqrt(RecoDecay::P(candidate.pxProng1(), candidate.pyProng1(), candidate.pzProng1())),                      \
```

Comment:
These calculations don't make sense to me.

Reply:
Indeed, thanks..